Godeps/_workspace/src/k8s.io/kubernetes/pkg/storage/etcd/etcd_helper.go

```diff
@@ -313,6 +313,20 @@ func (h *etcdHelper) Delete(ctx context.Context, key string, out runtime.Object)
 }
 
 // Implements storage.Interface.
+func (h *etcdHelper) DeleteDir(ctx context.Context, directory string) error {
+	if ctx == nil {
+		return errors.New("Context is nil")
+	}
+
+	directory = h.prefixEtcdKey(directory)
+
+	startTime := time.Now()
+	_, err := h.etcdKeysAPI.Delete(ctx, directory, &etcd.DeleteOptions{Dir: true})
+	metrics.RecordEtcdRequestLatency("delete", directory, startTime)
```

Comment:
i wanted to be able to distinguish delete directory differently than delete of item.